src/java/org/apache/cassandra/cql3/constraints/ConstraintFunction.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.constraints;
+
+
+import org.apache.cassandra.cql3.Operator;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.schema.ColumnMetadata;
+
+/**
+ * Interface to be implemented by functions that are executed as part of CQL constraints.
+ */
+public interface ConstraintFunction
```

Comment:
Currently this is only implemented by `LengthConstraint`. If we don't have any other immediate use-cases in mind we should move away from having an interface w/1 implementation, and if we _do_ have other use-cases in mind or coming down the pike, we should allude to that here IMO.

Reply:
I'd like to keep it. There are more constraints proposed on the CEP that I would like to implement, and this one was implemented to kick constraints off with enough value on its own.